### ISSUE TITLE
Static image wrongly mapped when update to new version in Android

### ISSUE THREAD

**Issue by anniewey** [facebook/react-native#32884]
### Description

The images are showing up but they are wrongly map all over the screen [refer:image]. It occurs whenever user installed new update over the old version. Only happen in Android release build, totally fine in debug mode. Refer to [screenshot](https://i.stack.imgur.com/IZBNu.jpg)

If user uninstall and reinstall, then the app appears fine. But it happen again when user install new update over it. I couldn't force user to uninstall and reinstall each time new update is released to store.

I'm using `require(./image/xx.png)` to pass in the `<Image>` source. 

Was wondering is it something went wrong during assets bundling? I've been debugging this issue for whole day but to no avail, hence creating this issue.

### Version

0.66.1

### Output of `npx react-native info`

```
System:
    OS: macOS 11.6
    CPU: (8) x64 Apple M1
    Memory: 43.58 MB / 8.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 17.3.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 8.3.0 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.0.1, iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0
    Android SDK:
      API Levels: 26, 28, 29, 30, 31
      Build Tools: 28.0.3, 29.0.2, 30.0.2, 30.0.3
      System Images: android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.6953283
    Xcode: 13.1/13A1030d - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_275 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.66.1 => 0.66.1 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### Steps to reproduce

1. `cd android && ./gradlew clean`
2. `./gradlew assembleRelease`
3. Install new APK over existing version
4. Image show up wrongly

### Snack, code example, screenshot, or link to a repository

**Example usage:**
```
const IMG_LOGO = require('../../../assets/image/auth/logo.png')

<Image source={IMG_LOGO} style={styles.imgLogo} resizeMode='contain'/>
```

**Comment by zhoumi1324** [comment]
请问你解决了吗  我也碰到了同样的问题， 男嗖！！

**Comment by anniewey** [comment]
@zhoumi1324 I couldn't find a permanent fix, temporarily using `uri` to load image instead.

`<Image source={{uri:'https://...'}} />`

**Comment by marcusbrandt** [comment]
Same issue here

**Comment by iamiota** [comment]
+1

**Comment by anniewey** [comment]
Fixed using solution from #32671. Downgrade fresco version from `2.6.0` to `2.5.0`

 `implementation 'com.facebook.fresco:animated-gif:2.5.0'`

**Comment by pierroo** [comment]
@anniewey can you please clarify where did you add that line exactly?
The link you shared they are using a different implementation, by having it in 

configurations.all {
    resolutionStrategy {
        force "com.facebook.fresco:fresco:2.5.0"
    }
}

and why animated-gif only?

**Comment by anniewey** [comment]
@pierroo i'm using `animated-gif` only in my project for the gif support as documented [here](https://reactnative.dev/docs/0.69/image#gif-and-webp-support-on-android). that line is in `android/app/build.gradle`

```
dependencies {
    implementation 'com.facebook.fresco:animated-gif:2.5.0' 
}
```

**Comment by SakeFabi** [comment]
I am still running into this - tried adding all variants that I can find here `configurations.all {
resolutionStrategy {
force "com.facebook.fresco:fresco:2.5.0"
}
}`

and `dependencies {
    implementation 'com.facebook.fresco:animated-gif:2.5.0' 
}` 
and 
`dependencies {
    implementation 'com.facebook.fresco:fresco:2.5.0' 
}` 
anybody another idea?

**Comment by efstathiosntonas** [comment]
this is fixed on `Fresco@3.0.0` https://github.com/facebook/fresco/issues/2667#issuecomment-1476119354
